docs/source/dataset_formats.mdx

```diff
@@ -209,6 +209,7 @@ Choosing the right dataset format depends on the task you are working on and the
 | [`RewardTrainer`]       | [Preference (implicit prompt recommended)](#preference) |
 | [`SFTTrainer`]          | [Language modeling](#language-modeling)                 |
 | [`XPOTrainer`]          | [Prompt-only](#prompt-only)                             |
+| [`GKDTrainer`]          | [Conversational](#conversational-dataset-format)        |                    |
```

Comment:
conversational is a *type*, not a *format*. I think GKD needs prompt only but you'll need to double check

Reply:
please also sort it alphabeticaly

Reply:
> I think GKD needs prompt-only but you'll need to double check

>no GKD needs prompts as well as full conversations in the "messages" key,

in [GKDTrainer docs](https://huggingface.co/docs/trl/v0.11.3/en/gkd_trainer#expected-dataset-format) the expected data format is described as:
The dataset should be formatted as a list of “messages” where each message is a list of dictionaries with the following keys:

- role: either system, assistant or user
- content: the message content

like:
```python
messages = [
    {"role": "user", "content": "Hello, how are you?"},
    {"role": "assistant", "content": "I'm doing great. How can I help you today?"},
    {"role": "user", "content": "I'd like to show off how chat templating works!"},
]
```

which is why I wrote it as **Conversational**

is the GKDTrainer docs incorrect?

Reply:
> is the GKDTrainer docs incorrect?

it's not incorrect but it has to be updated. Don't worry about that, we will make this clarification in a follow-up PR